providers/docker-machine.go

```diff
@@ -0,0 +1,173 @@
+package providers
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"strconv"
+	"strings"
+)
+
+var dockerMachineStorePath = os.Getenv("MACHINE_STORAGE_PATH")
+
+type DockerMachineProvider struct{}
+
+type dockerMachineConfig struct {
+	ConfigVersion int `json:"ConfigVersion,omitempty"`
+	Driver        struct {
+		IPAddress  string `json:"IPAddress"`
+		SSHKeyPair string `json:"SSHKeyPair"`
+		SSHKeyPath string `json:"SSHKeyPath"`
+		SSHPort    int    `json:"SSHPort"`
+		SSHUser    string `json:"SSHUser"`
+	} `json:"Driver,omitempty"`
+	HostOptions struct {
+		EngineOptions struct {
+			Labels []string `json:"Labels,omitempty"`
+		} `json:"EngineOptions,omitempty"`
+	}
+	Name string `json:"Name,omitempty"`
+}
+
+func (d *DockerMachineProvider) ListNodes(reader *bufio.Reader) ([]string, error) {
```

Comment:
The function doesn't exactly list nodes.. The name is a bit confusing. Can you please rename it to something more descriptive?